File: apps/codemap-search/tests/fixtures/navigation/go/worker.go
package checkout

import "context"

type Worker struct {
	queue     JobQueue
	processor Processor
	logger    Logger
}

func (w Worker) Tick(ctx context.Context) error {
	job, ok := w.queue.Next(ctx)
	if !ok {
		w.logger.Warn("no job")
		return nil
	}

	receipt, err := w.processor.Process(ctx, job)
	if err != nil {
		return err
	}
	return w.queue.Ack(ctx, receipt.ID)
}

File: apps/codemap-search/tests/fixtures/navigation/rust/worker.rs
use tracing::warn;

pub struct Worker<Q, P> {
    queue: Q,
    processor: P,
}

impl<Q, P> Worker<Q, P>
where
    Q: JobQueue,
    P: JobProcessor,
{
    pub async fn tick(&self) -> Result<(), Error> {
        let Some(job) = self.queue.next().await? else {
            warn!("no job");
            return Ok(());
        };

        if let Some(retry) = self.queue.retry_hint().await? {
            warn!("retrying {retry}");
        }

        let receipt = self.processor.process(job).await?;
        self.queue.ack(receipt.id).await
    }
}
